### PR TITLE
[MediaBundle] refactor large action method

### PR DESCRIPTION
**Pull request by delboy1978uk**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | https://github.com/Kunstmaan/KunstmaanBundlesCMS/issues/1691

This is only one of the several files requiring fixed, so the issue ticket should remain open

### CODE REVIEW

**Comment by delboy1978uk** [comment]
I'm going to leave the rest of the files Insight is complaining about, as Kevin & I have decided a proper refactor of the domain logic from the controllers into a service would be more appropriate than just breaking the methods down into more smaller ones.
Namely, 
	•	88 lines for MediaController::bulkUploadSubmitAction() in src/…/Controller/MediaController.php, line 140 
	•	83 lines for NodeAdminController::editAction() in src/…/Controller/NodeAdminController.php, line 833 
	•	56 lines for AdminListController::doEditAction() in src/…/Controller/AdminListController.php, line 203 
	•	48 lines for AdminListController::doAddAction() in src/…/Controller/AdminListController.php, line 103 
	•	32 lines for NodeAdminController::reorderAction() in src/…/Controller/NodeAdminController.php, line 757 
	•	28 lines for NodeAdminController::duplicateAction() in src/…/Controller/NodeAdminController.php, line 484 
	•	25 lines for NodeAdminController::deleteAction() in src/…/Controller/NodeAdminController.php, line 419 
	•	23 lines for AdminListController::doDeleteAction() in src/…/Controller/AdminListController.php, line 360 
	•	22 lines for NodeAdminController::addAction() in src/…/Controller/NodeAdminController.php, line 644 
	•	22 lines for NodeAdminController::revertAction() in src/…/Controller/NodeAdminController.php, line 559 
So please review and merge what we've got! 👍😀

File: src/Kunstmaan/MediaBundle/Controller/ChooserController.php [hunk 1]
```diff
@@ -76,47 +80,57 @@ public function chooserIndexAction(Request $request)
...
+    /**
+     * @param $linkChooser
+     * @param $cKEditorFuncNum
+     * @return null|string
+     */
+    private function getLinkChooserLink($linkChooser, $cKEditorFuncNum)
+    {
         $linkChooserLink = null;
         if (!empty($linkChooser)) {
             $params = array();
```
Comment (deZinc) [inline src/Kunstmaan/MediaBundle/Controller/ChooserController.php]: Can you change to []?

File: src/Kunstmaan/MediaBundle/Controller/FolderController.php [hunk 2]
```diff
@@ -14,88 +15,118 @@
...
-                return new RedirectResponse(
-                    $this->generateUrl(
-                        'KunstmaanMediaBundle_folder_show',
-                        array('folderId' => $folderId)
-                    )
-                );
-            }
-        }
+        $sub->setParent($this->folder);
+        $this->subForm  = $this->createForm(FolderType::class, $sub, array('folder' => $sub));
```
Comment (deZinc) [inline src/Kunstmaan/MediaBundle/Controller/FolderController.php]: Short notation for the arrays

File: src/Kunstmaan/MediaBundle/Controller/FolderController.php [hunk 3]
```diff
@@ -14,88 +15,118 @@
...
+        $this->emptyForm = $this->createEmptyForm();
+        $this->editForm = $this->createForm(FolderType::class, $this->folder, array('folder' => $this->folder));
```
Comment (deZinc) [inline src/Kunstmaan/MediaBundle/Controller/FolderController.php]: Same here

File: src/Kunstmaan/MultiDomainBundle/Controller/SiteSwitchController.php [hunk 4]
```diff
@@ -14,6 +14,17 @@
 
 class SiteSwitchController extends Controller
```
Comment (deZinc) [inline src/Kunstmaan/MultiDomainBundle/Controller/SiteSwitchController.php]: Needs a docblock

File: src/Kunstmaan/SeoBundle/Controller/Admin/SettingsController.php [hunk 5]
```diff
@@ -15,6 +16,21 @@
...
+    /** @var \Doctrine\Common\Persistence\ObjectRepository */
+    private $repo;
+
+    /** @var ObjectManager $em */
+    private $em;
+
+    /** @var bool $isSaved */
+    private $isSaved;
+
+    public function __construct()
```
Comment (deZinc) [inline src/Kunstmaan/SeoBundle/Controller/Admin/SettingsController.php]: Docblock

File: src/Kunstmaan/SeoBundle/Controller/Admin/SettingsController.php [hunk 6]
```diff
@@ -26,45 +42,51 @@ class SettingsController extends BaseSettingsController
     public function robotsSettingsAction(Request $request)
```
Comment (deZinc) [inline src/Kunstmaan/SeoBundle/Controller/Admin/SettingsController.php]: doc....block :-p

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php [hunk 7]
```diff
@@ -27,10 +33,22 @@ class TranslatorController extends AdminListController
...
      */
     private $adminListConfigurator;
 
+    /** @var ObjectManager $em */
+    private $em;
+
+    /** @var TranslationRepository $repo */
+    private $repo;
+
+    public function __construct()
```
Comment (deZinc) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php]: d.......

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php [hunk 8]
```diff
@@ -51,10 +69,7 @@ public function indexAction(Request $request)
...
 
         if (!$cacheFresh && !$debugMode) {
-            $this->addFlash(
-                FlashTypes::INFO,
-                $this->get('translator')->trans('settings.translator.not_live_warning')
-            );
+            $this->addFlash(FlashTypes::INFO, $this->get('translator')->trans('settings.translator.not_live_warning'));
         }
 
         return array(
```
Comment (deZinc) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php]: [] notation would be better here.

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php [hunk 9]
```diff
@@ -78,52 +93,40 @@ public function indexAction(Request $request)
...
+                $this->em->flush();
+                $this->addFlash(FlashTypes::SUCCESS, $this->get('translator')->trans('settings.translator.succesful_added'));
                 $indexUrl = $configurator->getIndexUrl();
+                $params = isset($indexUrl['params']) ? $indexUrl['params'] : [];
 
-                return new RedirectResponse($this->generateUrl(
-                    $indexUrl['path'],
-                    isset($indexUrl['params']) ? $indexUrl['params'] : array()
-                ));
+                return new RedirectResponse($this->generateUrl($indexUrl['path'], $params));
```
Comment (deZinc) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php]: Per your own request, put the generateUrl in a var.

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php [hunk 10]
```diff
@@ -78,52 +93,40 @@ public function indexAction(Request $request)
...
         }
 
-        return array(
-            'form' => $form->createView(),
-            'adminlistconfigurator' => $configurator
-        );
+        return ['form' => $form->createView(), 'adminlistconfigurator' => $configurator];
+    }
+
+    private function addLocales(TranslationModel $translation)
```
Comment (deZinc) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php]: doc

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php [hunk 11]
```diff
@@ -140,16 +143,49 @@ public function addAction(Request $request, $keyword = '', $domain = '', $locale
...
+            }
+        }
+
+        return ['form' => $form->createView(), 'translation' => $translation, 'adminlistconfigurator' => $configurator];
+    }
 
-        $translations = $em->getRepository('KunstmaanTranslatorBundle:Translation')->findBy(array('translationId' => $id));
+    /**
+     * @param $id
+     * @return Translation[]
```
Comment (deZinc) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php]: Can you add the exception in the docblock.

File: src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php [hunk 12]
```diff
@@ -248,9 +255,9 @@ public function setAdminListConfigurator($adminListConfigurator)
     }
 
     /**
-     * @return AbstractAdminListConfigurator
+     * @return TranslationAdminListConfigurator
```
Comment (deZinc) [inline src/Kunstmaan/TranslatorBundle/Controller/TranslatorController.php]: Exception in the docblock

**Comment by delboy1978uk** [comment]
Should I squash these commits?

**Comment by deZinc** [comment]
@delboy1978uk I guess so, for the squashing, it are quiet a bit of commits otherwise.

File: src/Kunstmaan/DashboardBundle/Controller/GoogleAnalyticsController.php [hunk 13]
```diff
@@ -19,63 +29,68 @@ class GoogleAnalyticsController extends Controller
...
+        $params['disableGoals'] = $analyticsConfigRepository->findFirst()->getDisableGoals();
+        $params['segments'] = $analyticsConfigRepository->findFirst()->getSegments();
+        $date = $analyticsConfigRepository->findFirst()->getLastUpdate();
+        $params['last_update'] = ($date) ? $date->format('d-m-Y H:i') : 'never';
+
+        return $params;
+    }
+
+    /**
+     * @param null $segmentId
```
Comment (deZinc) [inline src/Kunstmaan/DashboardBundle/Controller/GoogleAnalyticsController.php]: Null or int I guess?

Comment (delboy1978uk) [reply src/Kunstmaan/DashboardBundle/Controller/GoogleAnalyticsController.php]: docblock fixed, and commits squashed

**Comment by delboy1978uk** [comment]
Debugging the test right now!

File: src/Kunstmaan/DashboardBundle/Controller/GoogleAnalyticsController.php [hunk 14]
```diff
@@ -19,63 +29,68 @@ class GoogleAnalyticsController extends Controller
...
-            $params['last_update'] = "never";
+
+    /**
+     * @param ConfigHelper $configHelper
+     * @param array $params
+     * @return Response
+     */
+    private function connectToGoogle(ConfigHelper $configHelper, array $params)
+    {
+        if ($this->getParameter('google.api.client_id') != '' && $this->getParameter('google.api.client_secret') != '' && $this->getParameter('google.api.dev_key') != '' ) {
```
Comment (sandergo90) [inline src/Kunstmaan/DashboardBundle/Controller/GoogleAnalyticsController.php]: Can you make it strict ?

File: src/Kunstmaan/DashboardBundle/Controller/GoogleAnalyticsController.php [hunk 15]
```diff
@@ -115,27 +130,44 @@ public function setTokenAction(Request $request)
...
 
-        $params = array();
-        $configHelper = $this->container->get('kunstmaan_dashboard.helper.google.analytics.config');
-
         if (null !== $request->request->get('accounts')) {
             return $this->redirect($this->generateUrl('kunstmaan_dashboard'));
         }
 
+        /** @var ConfigHelper $configHelper */
+        $configHelper = $this->container->get('kunstmaan_dashboard.helper.google.analytics.config');
```
Comment (sandergo90) [inline src/Kunstmaan/DashboardBundle/Controller/GoogleAnalyticsController.php]: Can you remove the container usage but just do $this->get() ?

File: src/Kunstmaan/UserManagementBundle/Controller/UsersController.php [hunk 16]
```diff
@@ -171,51 +165,84 @@ public function editAction(Request $request, $id)
...
 
-        if ($tabPane) {
-            $params = array_merge($params, array('tabPane' => $tabPane));
+    /**
+     * @return string
+     */
+    private function getRequiredRole($id)
+    {
+        // The logged in user should be able to change his own password/username/email and not for other users
+        if ($id == $this->get('security.token_storage')->getToken()->getUser()->getId()) {
```
Comment (sandergo90) [inline src/Kunstmaan/UserManagementBundle/Controller/UsersController.php]: Can be simplified to return 'ROLE_ADMIN' and no else statement but return 'ROLE_SUPER_ADMIN'

File: src/Kunstmaan/UserManagementBundle/Controller/UsersController.php [hunk 17]
```diff
@@ -171,51 +165,84 @@ public function editAction(Request $request, $id)
...
-                $userManager = $this->container->get('fos_user.user_manager');
-                $userManager->updateUser($user, true);
+    /**
+     * @param Request $request
+     * @param Form $form
+     * @param User $user
+     * @param TabPane|null $tabPane
+     * @return null|RedirectResponse
+     */
+    private function handleEditForm(Request $request, Form $form, User $user, TabPane $tabPane = null)
```
Comment (sandergo90) [inline src/Kunstmaan/UserManagementBundle/Controller/UsersController.php]: Maybe make function protected?

File: src/Kunstmaan/UserManagementBundle/Controller/UsersController.php [hunk 18]
```diff
@@ -26,6 +31,9 @@
  */
 class UsersController extends BaseSettingsController
 {
+    /** @var ObjectManager $em */
```
Comment (sandergo90) [inline src/Kunstmaan/UserManagementBundle/Controller/UsersController.php]: Use EntityManagerInterface

File: src/Kunstmaan/SeoBundle/Controller/Admin/SettingsController.php [hunk 19]
```diff
@@ -12,59 +13,88 @@
...
+/**
+ * Class SettingsController
+ * @package Kunstmaan\SeoBundle\Controller\Admin
+ */
 class SettingsController extends BaseSettingsController
 {
+    /** @var \Doctrine\Common\Persistence\ObjectRepository */
+    private $repo;
+
+    /** @var ObjectManager $em */
```
Comment (sandergo90) [inline src/Kunstmaan/SeoBundle/Controller/Admin/SettingsController.php]: Use EntityManagerInterface

File: src/Kunstmaan/NodeBundle/Controller/SlugController.php [hunk 20]
```diff
@@ -49,56 +78,88 @@ public function slugAction(Request $request, $url = null, $preview = false)
...
+
+    /**
+     * @param NodeTranslation $nodeTranslation
+     * @param $url
+     * @param EntityInterface $entity
+     * @param NodeMenu $nodeMenu
+     * @param SlugSecurityEvent $securityEvent
+     * @return RenderContext
+     */
+    private function getRenderContext(NodeTranslation $nodeTranslation, $url, EntityInterface $entity, NodeMenu $nodeMenu, SlugSecurityEvent $securityEvent)
```
Comment (sandergo90) [inline src/Kunstmaan/NodeBundle/Controller/SlugController.php]: I think this functions need to be protected better

File: src/Kunstmaan/NodeBundle/Controller/SlugController.php [hunk 21]
```diff
@@ -49,56 +78,88 @@ public function slugAction(Request $request, $url = null, $preview = false)
...
+    {
+        $this->eventDispatcher = $this->get('event_dispatcher');
```
Comment (sandergo90) [inline src/Kunstmaan/NodeBundle/Controller/SlugController.php]: It's strange that the eventdispatcher is collected from container here. Maybe move it to correct place.

**Comment by sandergo90** [comment]
It's also not a bugfix so should be rebased against master

**Comment by Devolicious** [comment]
Closing this due to inactivity

**Pull request closed without merging**